tests/withDynamicResponsiveProps.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { render } from 'react-testing-library'
+import OIOResponsiveObjectPropType from '../src/utils/PropType'
+import generateResponsiveObject from '../src/utils/generateResponsiveObject'
+import withDynamicResponsiveProps from '../src/utils/withDynamicResponsiveProps'
+import { breakpointKeys } from '../config/constants'
+
+// Test Fixture
+
+const ResponsiveText = ({ breakpoint, fontSize, fontWeight, lineHeight, value }) => (
+   <span style={{
+      fontSize: fontSize && fontSize[breakpoint],
+      fontWeight: fontWeight && fontWeight[breakpoint],
+      lineHeight: lineHeight && lineHeight[breakpoint]
+   }}>
+      {value}
+   </span>
+)
+
+ResponsiveText.propTypes = {
+   breakpoint: PropTypes.oneOf(breakpointKeys).isRequired,
+   fontSize: OIOResponsiveObjectPropType.isRequired,
+   fontWeight: OIOResponsiveObjectPropType,
+   lineHeight: OIOResponsiveObjectPropType,
+   value: PropTypes.string.isRequired
+}
+
+ResponsiveText.defaultProps = {
+   fontWeight: null,
+   lineHeight: null
+}
+
+// Tests
+
+test('Will work when creating dynamic props from props at each breakpoint', () => {
+   // Make the line height and font weight proportional to the font size & screen size
+   const DecoratedResponsiveText = withDynamicResponsiveProps((props, breakpoint) => {
+      const fontSize = parseInt(props.fontSize[breakpoint], 10)
+      const { fontWeightMultiplier, lineHeightMultiplier } = props
+      const baseCharCode = 'a'.charCodeAt(0)
+      const currentCharCode = breakpoint.charCodeAt(0)
+      // eslint-disable-next-line no-mixed-operators
+      const breakpointDiff = currentCharCode - baseCharCode + 1
+      return ({
+         fontWeight: fontSize * fontWeightMultiplier * breakpointDiff,
+         lineHeight: fontSize * lineHeightMultiplier * breakpointDiff
+      })
+   })(ResponsiveText)
+
+   DecoratedResponsiveText.defaultProps = {
+      fontWeightMultiplier: 1,
+      fontSize: generateResponsiveObject('9[ab] 10[cd] 11[ef]'),
+      lineHeightMultiplier: 0.5
+   }
+
+   const { container } = render(
+      <React.Fragment>
+         <DecoratedResponsiveText breakpoint="a" value="A [fw=9*1*1] [lh=9*0.5*1]" />
```

Comment:
Not sure if this matters or not, but I was pretty confused cause I didn't realize that fw and lh wear stood for font weight and line height. Maybe we can make a little more verbose for clarity?

Reply:
Yeah it's just because of the linting max-len rule that I didn't cuz then. I'll do something.